proxy/src/control/destination/background.rs

```diff
@@ -72,6 +88,18 @@ struct DestinationSet<T: HttpService<ResponseBody = RecvBody>> {
     responders: Vec<Responder>,
 }
 
+/// The state needed to bind a new controller client stack.
+///
+/// This implements `watch_service::Rebind`, so it can be used to rebind the
+/// client when the TLS client configuration changes.
+struct BindClient {
+    identity: Conditional<tls::Identity, tls::ReasonForNoTls>,
+    host_and_port: HostAndPort,
+    dns_resolver: dns::Resolver,
+    scheme: http::uri::Scheme,
+    authority: http::uri::Authority,
```

Comment:
`authority` is just a copy of `host_and_port` with a different type. I think `scheme` supposed to be `https` when we use TLS and `http` when we don't.

Since we'll rebind (construct a new `AddOrigin`) very rarely, I think it would be simpler & clearer if we simply calculated `scheme` and `authority` in local variables right before we construct the `AddOrigin`, and avoid storing them in this struct.

Otherwise, we should at least document these relationships since it tool a non-trivial amount of time for me to figure them out to write this comment.

Reply:
Okay, I'll undo that change 👍